src/logic/safe/store/actions/fetchSafe.ts

```diff
@@ -23,18 +23,21 @@ export const buildSafe = async (safeAddress: string): Promise<SafeRecordProps> =
   // setting `loadedViaUrl` to false, as `buildSafe` is called on safe Load or Open flows
   const safeInfo: Partial<SafeRecordProps> = { address, loadedViaUrl: false }
 
-  const [remote, localSafeInfo] = await allSettled<[SafeInfo | null, SafeRecordProps | undefined | null]>(
-    getSafeInfo(safeAddress),
+  const [remote, localSafeInfo] = await Promise.all([
+    getSafeInfo(safeAddress).catch((err) => {
+      logError(Errors._605, err.message)
+      return null
+    }),
     getLocalSafe(safeAddress),
-  )
+  ])
 
   // remote (client-gateway)
   const remoteSafeInfo = remote ? await extractRemoteSafeInfo(remote) : {}
 
   // update owner's information
   const owners = buildSafeOwners(remote?.owners, localSafeInfo?.owners)
 
-  return { ...localSafeInfo, ...safeInfo, ...remoteSafeInfo, owners } as SafeRecordProps
+  return { ...(localSafeInfo || {}), ...safeInfo, ...remoteSafeInfo, owners } as SafeRecordProps
```

Comment:
I think because the load function can return undefined (default value works) or null(default value doesn't work), at least this is what function signature mentions